packages/blockchain/algo/src/lib/services/algo.tx.ts

```diff
@@ -499,16 +795,104 @@ export const algoTx = (args: { algoWeb: AlgoWeb }) => {
-          }),
+        signedTransaction: async (
+          body: TransferAlgo | TransferAlgoKMS,
+          testnet = false,
+          provider?: string,
+        ) => {
+          const txData = await prepareSignedTransaction(body, testnet, args.algoWeb, provider)
+          if ('signatureId' in body) {
+            return ApiServices.blockchain.algo.algorandBlockchainTransfer(JSON.parse(txData))
+          }
+          return AlgorandService.algorandBroadcast({
+            txData,
+          })
+        },
+      },
+    },
+
+    virtualAccount: {
+      send: {
+        /**
+         * Send ALGO transaction from Tatum Ledger account to the blockchain. This method broadcasts signed transaction to the blockchain.
+         * This operation is irreversible.
+         * @param body content of the transaction to broadcast
+         * @param testnet if the algorand node is testnet or not
+         * @param provider url of the algorand server endpoint for purestake.io restapi
+         * @returns transaction id of the transaction in the blockchain or id of the withdrawal, if it was not cancelled automatically
+         */
+        signedTransaction: async (
+          body: ApiTransferAlgo | ApiTransferAlgoKMS | ChainTransferAlgoErc20 | ChainTransferAlgoErc20KMS,
+          testnet = false,
+          provider?: string,
+        ): Promise<SendOffchainResponse> => {
+          let txData
+
+          const { fee, ...withdrawal } = body as ApiTransferAlgoKMS
+
+          const { senderAccountId } = withdrawal
+          const account = await AccountService.getAccountByAccountId(senderAccountId)
+
+          if (account.currency === Currency.ALGO) {
+            txData = await prepareSignedTransaction(
+              body as ApiTransferAlgo | ApiTransferAlgoKMS,
+              testnet,
+              args.algoWeb,
+              provider,
+            )
+          } else {
+            txData = await prepareTransferFTSignedTransaction(
+              {
+                from: (
+                  await AlgorandService.algorandGenerateAddress(
+                    (body as ChainTransferAlgoErc20).fromPrivateKey,
+                  )
+                )?.address,
+                fromPrivateKey: (body as ChainTransferAlgoErc20).fromPrivateKey,
+                signatureId: (body as ChainTransferAlgoErc20KMS).signatureId,
+                to: (body as ChainTransferAlgoErc20 | ChainTransferAlgoErc20KMS).to,
+                amount: body.amount,
+                contractAddress: (body as ChainTransferAlgoErc20 | ChainTransferAlgoErc20KMS).contractAddress,
+              },
+              testnet,
+              args.algoWeb,
+              provider,
+            )
+          }
+
+          const { id } = await ApiServices.offChain.withdrawal.storeWithdrawal({
+            ...withdrawal,
+            fee: new BigNumber(fee || '0.001').toString(),
+          })
+
+          if ('signatureId' in body) {
```

Comment:
this should be moved to the beginning of the method and should call https://apidoc.tatum.io/tag/Blockchain-operations#operation/AlgoTransfer directly without any local logic